### ISSUE TITLE
Investigate smaller (filesize) thumbnails

### ISSUE THREAD

**Issue by hyena** [Weasyl/weasyl#6]
The large thumbnails are pretty but they may be contributing to slow site load speeds. Investigate whether jpg thumbs for .png files are worth it without sacrificing too much quality. Images that use transparency will still have to be png's or gifs.

Cursory investigation suggests that identification of transparency in the alpha channels may be expensive under sanpera as it stands: Potentially a loop over all the pixel objects. We could expose some more ImageMagick functionality if necessary.

**Comment by Syfaro** [comment]
I'd like to point out for those of us with retina displays, current thumbnails can already be a little blurry. I hope that switching to a lower quality image wouldn't make it worse.

**Comment by Arcaena** [comment]
Yeah, I've got some pretty high dpi displays and I'd be concerned about potentially worse thumbs.

**Comment by taedixon** [comment]
Perhaps it could be worthwhile for users to be able to opt-in to 'high-resolution' thumbnails (premium feature?? :sweat_smile: ) but those on mobile or rural networks the added load times can be frustrating

**Comment by strain-113** [comment]
Multiple formats are possible. The design of thumbnails actually was done in such a way as to permit multiple thumbnails with various attributes (remember how you can disable user created thumbnails for site-generated ones?), although fixing every place in the code that assumes a single thumbnail will be non-trivial.

**Comment by kfkitsune** [comment]
So, here are my findings for this.

https://gist.github.com/kfkitsune/43464081d15ea709488db6e8b36da814
http://www.mediafire.com/download/0femd55wyq1wlbf/thumbnailTests.tar.xz

Summary:

- JPG thumbnails that are at 100/99 percent quality can be reduced to 95% without much loss of visual quality, and a reduced filesize from 163033 bytes to 62227 bytes. (Trial 1 folder of linked tar.xz)

- PNG files can be similarly compressed to 95% JPG. Trial 2: 60823 bytes versus 152595 bytes for the original PNG thumbnail. Trial 3: 58653 bytes compressed versus 168615 bytes for the original PNG thumbnail.

- PNG files converted to JPG at 100% quality showed a slight reduction in file size. Trial 2: 135819 versus 152595 original. Trial 3: 127199 versus 168615 original.

- Imagemagick's ``identify -verbose

`` command can (rather easily) be used to determine if there is an alpha channel. Said section is called out directly in the "Channel statistics" section, with an "Alpha" subsection.

I also included all the conversions I did in a single captured image within each subfolder of the linked archive. Further, I included a blow-up of the original versus the 95% compressed version for your visual enjoyment/comparison.
